lib/linguist/languages.yml

```diff
@@ -2832,6 +2832,9 @@ Lex:
   extensions:
   - ".l"
   - ".lex"
+  filenames:
+  - Lexer.x
```

Comment:
I don't believe Linguist matches filenames (or extensions) case-sensitively, so there's no need to include a capitalised variant.

~~~diff
+  filenames:
-  - Lexer.x
+  - lexer.x
~~~

Reply:
@Alhadis Are you sure that is true. I listed both because when I was looking through `languages.yml` I noticed that both `Makefile` and `makefile` for Makefile filenames. 

Furthermore I updated the PR, with the suggested changes and though some of the tests fail because the sample is named `Lexer.x` and not `lexer.x` (the only filename that appears in `languages.yml`. The classification also fails and misclassifies `Lexer.x` as logos. This was not the case when `Lexer.x` was in `languages.yml` (as was the case in the earlier state of the PR)


```1) Failure:
TestClassifier#test_classify_ambiguous_languages [/home/runner/work/linguist/linguist/test/test_classifier.rb:56]:
/home/runner/work/linguist/linguist/samples/Lex/filenames/Lexer.x
[["Logos", -20405.75990513595], ["Linker Script", -20545.87827297192], ["RPC", -20899.45156547006], ["DirectX 3D File", -25058.54557573806]].
Expected: "Lex"
  Actual: "Logos"

  2) Failure:
TestFileBlob#test_language [/home/runner/work/linguist/linguist/test/test_file_blob.rb:648]:
Lexer.x.
Expected: "Lex"
  Actual: "Logos"

  3) Failure:
TestSamples#test_filename_listed [/home/runner/work/linguist/linguist/test/test_samples.rb:49]:
/home/runner/work/linguist/linguist/samples/Lex/filenames/Lexer.x isn't listed as a filename for Lex in languages.yml.
Expected ["lexer.x"] to include "Lexer.x".

  4) Failure:
TestBlob#test_language [/home/runner/work/linguist/linguist/test/test_blob.rb:253]:
Lexer.x.
Expected: "Lex"
```

Failure 1) specifically

Reply:
What the hell… 😕 Okay, that was unexpected. Leave it then, we can figure this out some other time.